src/streamer.ts

```diff
@@ -0,0 +1,36 @@
+import { S3Client } from '@aws-sdk/client-s3';
+import { listBlocks, fetchStreamerMessage } from './s3fetchers';
+import { LakeConfig, BlockHeight, StreamerMessage } from './types';
+
+export async function startStream(
+  config: LakeConfig,
+  onStreamerMessageReceived: (data: StreamerMessage) => Promise<void>
+) {
+  const s3Client = new S3Client({ region: config.s3RegionName });
+
+  let lastProcessedBlockHash: string;
+  let startFromBlockHeight = config.startBlockHeight;
+  let queue: Promise<void>[] = [];
+
+  while (true) {
+    const blockHeights = await listBlocks(s3Client, config.s3BucketName, startFromBlockHeight);
+    if (blockHeights.length > 0) {
+      startFromBlockHeight = blockHeights[blockHeights.length - 1];
+    } else {
+      setTimeout(() => {}, 2000);
```

Comment:
If it is intended to make a pause before next fetch - it should be something like this:
```ts
// utils.ts
const wait = (pause: number) => new Promise((resolve) => setTimeout(resolve, pause))

// index.ts
await wait(2000);
```